data/model/oci/tag.py

```diff
@@ -116,6 +116,19 @@ def get_child_manifests(repo_id: int, manifest_id: int):
     )
 
 
+def get_child_manifests_aggregate_size(repo_id: int, manifest_id: int):
+    return (
+        Manifest.select(fn.Sum(Manifest.layers_compressed_size))
+        .join(ManifestChild, on=(Manifest.id == ManifestChild.child_manifest))
+        .where(
+            (ManifestChild.repository == repo_id)
+            & (ManifestChild.manifest == manifest_id)
+            & (Manifest.layers_compressed_size.is_null(False))
+        )
+        .scalar()
```

Comment:
The reason I haven't approved this yet is concerns over this query. The list tags endpoint is our most requested endpoint so this will be ran many, many times. I'm thinking it shouldn't be an issue because the look ups will happen via indexed manifest ID columns but the `layers_compressed_size` is not indexed. Can we see the execution plan of this query via `EXPLAIN` or another method? We need to make sure this won't cause significant performance issues if deployed.

Reply:
This query will get the aggregate if the immediate child manifests, but what about other manifest lists. Wouldn't you need to recursively go through manifest lists to get an accurate size?